packages/@aws-cdk/aws-ecr/lib/public-repository.ts

```diff
@@ -0,0 +1,428 @@
+    }
+    const rule = new events.Rule(this, id, options);
+    rule.addEventPattern({
+      source: ['aws.ecr-public'],
+      resources: [this.publicRepositoryArn],
+    });
+    rule.addTarget(options.target);
+    return rule;
+  }
+
+  public grant(grantee: iam.IGrantable, ...actions: string[]) {
+    return iam.Grant.addToPrincipalAndResource({
+      grantee,
+      actions,
+      resourceArns: [this.publicRepositoryArn],
+      resourceSelfArns: [],
+      resource: this,
+    });
+  }
+
+  public grantPush(grantee: iam.IGrantable) {
+    const ret = this.grant(grantee, ...PublicRepositoryBase.PUSH_IAM_ACTIONS);
+
+    iam.Grant.addToPrincipal({
+      grantee,
+      actions: ['ecr-public:GetAuthorizationToken', 'sts:GetServiceBearerToken'],
+      resourceArns: ['*'],
+      scope: this,
+    });
+
+    return ret;
+  }
+}
+
+/**
+ * Options for the ECR Public Repository construct.
+ */
+export interface PublicRepositoryProps extends BaseRepositoryProps {
+  /**
+   * Name for this repository
+   *
+   * @default Automatically generated name.
+   */
+  readonly publicRepositoryName?: string;
+
+  /**
+   * A short description of the contents of the repository.
+   * This text appears in both the image details and also when searching for repositories on the Amazon ECR Public Gallery.
+   *
+   * @default - No description.
+   */
+  readonly description?: string;
+
+  /**
+   * A detailed description of the contents of the repository.
+   * It's publicly visible in the Amazon ECR Public Gallery. The text must be in markdown format.
+   *
+   * @default - No about text.
+   */
+  readonly about?: string;
+
+  /**
+   * Detailed information about how to use the contents of the repository. It's publicly visible in the Amazon ECR Public Gallery.
+   * The usage text provides context, support information, and additional usage details for users of the repository.
+   * The text must be in markdown format.
+   *
+   * @default - No usage text.
+   */
+  readonly usage?: string;
+
+  /**
+   * The system architecture that the images in the repository are compatible with.
+   *
+   * @default - No system architecture tags.
+   */
+  readonly architectures?: Architecture[];
+
+  /**
+   * The operating systems that the images in the repository are compatible with.
+   *
+   * @default - No operating system tags.
+   */
+  readonly operatingSystems?: OperatingSystem[];
+
+  /**
+   * Custom resource policy document to attach to the Public Repository.
+   *
+   * @default - No policy document. Only the repository owner has access to manage the repository.
+   */
+  readonly resourcePolicy?: iam.PolicyStatement[];
+}
+
+/**
+ * ECR Public Repository attributes.
+ */
+export interface PublicRepositoryAttributes {
+
+  /**
+   * The name of the repository.
+   */
+  readonly publicRepositoryName: string;
+
+  /**
+   * The ARN of the repository.
+   */
+  readonly publicRepositoryArn: string;
+}
```

Comment:
We can re-use the `RepositoryAttributes` interface since they are the same.

Reply:
I cannot do this because of linter rule violation.